docs/recipes/UsageWithTypescript.md

```diff
@@ -230,6 +230,26 @@ const rootReducer = combineReducers({
 export type RootState = ReturnType<typeof rootReducer>
 ```
 
+### Type Checking Middlewares
+
+A middleware is a higher-order function that composes a dispatch function to return a new dispatch function.
+To create a type checked middleware function, we can leverage the `Middleware` interface provided by Redux together with our store type (as defined in the previous example):
+
+```ts
+// src/middleware/example.ts
+
+import { Middleware } from 'redux'
+
+import { RootState } from '../store'
+
+export const exampleMiddleware: Middleware<
+  {},
```

Comment:
We may want to briefly explain what this generic is for, as it's not that obvious from this example.

Reply:
Sure, I can add a note. Would you prefer it in a standalone commit or squashed?

Reply:
Either is fine. We're going to squash it all anyways when it gets merged.

Reply:
👍  3c0023ac